zerver/tornado/views.py

```diff
@@ -137,64 +125,97 @@ def get_events(request: HttpRequest, user_profile: UserProfile) -> HttpResponse:
     return get_events_backend(request, user_profile)
 
 
-@has_request_variables
+class UserClient(BaseModel):
+    id: int
+    name: Annotated[str, StringConstraints(max_length=30)]
+
+    @model_validator(mode="before")
+    @classmethod
+    def convert_term(cls, elem: str) -> dict[str, Any]:
+        client = get_client(elem)
+        return {"id": client.id, "name": client.name}
```

Comment:
This might work but is a bit weird; I think you might just want to have just have the `Client` type itself be what comes out.

Reply:
Pydantic cannot generate the schema for Django models. Even with a custon validator specified, it still complains:
```
Unable to generate pydantic-core schema for <class 'zerver.models.clients.Client'>. Set `arbitrary_types_allowed=True` in the model_config to ignore this error 
```

Reply:
OK, we can live with this version for now.